src/platform_impl/macos/util/mod.rs

```diff
@@ -79,6 +79,35 @@ impl Clone for IdRef {
     }
 }
 
+macro_rules! trace_scope {
+    ($s:literal) => {
+        $crate::platform_impl::platform::util::TraceGuard::new(module_path!(), $s)
+    };
+}
```

Comment:
Are you absolutely certain that this will drop the `TraceGuard` at the end of the scope instead of being dropped immediately when used as `trace_scope!();`? I'm pretty sure the `scopeguard` crate uses a `let` binding for this.

Reply:
No, you're definitely right, I intended to add that but totally forgot!

Reply:
Good catch. Now I actually tried printing some text after the `trace_scope` and the printed text showed up between the Triggered and the Completed traces, so great!